src/TranslationResult.h

```diff
@@ -8,101 +8,178 @@
+  }
 
-  /* Return the translated text. */
-  const std::string &getTranslatedText() const { return translatedText; }
+  void set_alignments(std::vector<Alignment> &&alignments) {
+    alignments_ = std::move(alignments);
+  }
+
+  // Access to source as a whole.
+  const std::string &source() const { return source_.blob; };
+
+  // Access to translation as a whole.
+  const std::string &translation() const { return translation_.blob; };
+
+  /// Returns the number of units that are translated. To simplify thinking can
+  /// be thought of as number of sentences (equal in source and target by nature
+  /// of the task).
+  const size_t size() const { return source_.numSentences(); }
 
-  /* Return the Quality scores of the translated text. */
-  const QualityScore &getQualityScore() const { return qualityScore; }
+  const size_t numSourceWords(size_t sentenceIdx) {
+    auto terminals = source_.annotation.sentenceTerminals(sentenceIdx);
+    return terminals.second - terminals.first + 1;
+  }
 
-  /* Return the Sentence mappings (information regarding how individual
-   * sentences of originalText map to corresponding translated sentences in
-   * translatedText).
-   */
-  const SentenceMappings &getSentenceMappings() const {
-    return sentenceMappings;
+  const size_t numTranslationWords(size_t sentenceIdx) {
+    auto terminals = translation_.annotation.sentenceTerminals(sentenceIdx);
+    return terminals.second - terminals.first + 1;
   }
 
+  /// Allows access to a sentence mapping, when accesed iterating through
+  /// indices using size(). Intended to substitute for getSentenceMappings.
+
+  /// Access to source sentences.
+  std::string_view source_sentence(size_t idx) const {
+    return source_.annotation.sentence(idx);
+  };
+
+  /// Access to translated sentences.
+  std::string_view translated_sentence(size_t idx) const {
+    return translation_.annotation.sentence(idx);
+  };
+
+  /// Access to word level annotation in source.
+  std::string_view source_word(size_t sentenceIdx, size_t wordIdx) const {
+    return source_.annotation.word(sentenceIdx, wordIdx);
+  }
+
+  /// Access to word level annotation in translation
+  std::string_view translation_word(size_t sentenceIdx, size_t wordIdx) const {
+    return translation_.annotation.word(sentenceIdx, wordIdx);
+  }
+
+  /// Both quality and alignment will require tapping into token level.
+  /// information. wordIdx information required to extract corresponding
+  /// word-unit is embedded internally as indices in vectors (Quality) or
+  /// sparse-matrix constructions with indices (Alignment).
+
+  /// Returns quality score associated with the i-th pair.
+  const Quality &quality(size_t sentenceIdx) const {
+    return scores_[sentenceIdx];
+  }
+
+  /// Returns alignment information for the i-th pair.
+  const Alignment &alignment(size_t sentenceIdx) const {
+    return alignments_[sentenceIdx];
+  }
+
+  const std::string_view getOriginalText() const {
```

Comment:
Open an issue for this?

Reply:
Wait. Why is this function deprecated in favor of `source()` API. Both of them do the same task. Right?

The same question goes for deprecating `getTranslatedText()` API below.

Reply:
Issue: https://github.com/browsermt/bergamot-translator/issues/54

Reply:
Is this all a style argument?  Sigh.  For what it's worth, I think source() is a cleaner name than getOriginalText().  But I think having both with some std::cerr printout on one of them is worse than either option.

Reply:
Consistency is the argument. I'm still unaware of what bergamot-translator-app's true purpose is, I use it for regression-tests (std::cerr / std::cout). This is why `source()` is the proposal, `getOriginalText()` being used and that tests still work is that I didn't break anything. 

In any case, what I'm trying to say is: Solve #54 (this is a problem, nick's bytebuffer changes uses snakecase, I used camelcase, I'm not sure what to call what abhi uses) fixes for this will happen here.

Reply:
@jerinphilip I agree on having a common coding style/convention to be followed. However, I am for handling that issue in a separate PR where we can refactor all files and functions to follow one convention that we decide.

Let's not make this refactoring a part of this PR please 👍